xdsl/transforms/memref_stream_interleave.py

```diff
@@ -155,8 +147,8 @@ class MemrefStreamInterleavePass(ModulePass):
     Tiles the innermost parallel dimension of a `memref_stream.generic`.
     If specified, the `pipeline-depth` parameter specifies the number of operations in the
     resulting body that should be executed concurrently.
-    The pass will search through possible factors, starting with `pipeline-depth`, going
-    up to `pipeline-depth * 2`, and then down to 1, stopping at the first one.
+    The pass will select the largest factor of the corresponding bound smaller than
+    `pipeline-depth * 2`.
     The search range is bound by `pipeline-depth * 2` as very large interleaving factors
     would cause too much register pressure, potentially running out of registers.
```

Comment:
```suggestion
    The search range is bound by `pipeline-depth * 2` as very large interleaving factors
    can potentially increase register pressure or even exhaust all available registers.
```

I know this isn't directly related, but the wording has been annoying me for a bit.
Take it or leave it :)